Admin.js

```diff
@@ -114,561 +141,775 @@ var Admin = {
+            const protoName = container.attr("data-prototype-name") || "__name__";
             // Set field id
-            var idRegexp = new RegExp(container.attr('id')+'_'+protoName,'g');
-            proto = proto.replace(idRegexp, container.attr('id')+'_'+counter);
+            const idRegexp = new RegExp(`${container.attr("id")}_${protoName}`, "g");
+            proto = proto.replace(idRegexp, `${container.attr("id")}_${counter}`);
 
             // Set field name
-            var parts = container.attr('id').split('_');
-            var nameRegexp = new RegExp(parts[parts.length-1]+'\\]\\['+protoName,'g');
-            proto = proto.replace(nameRegexp, parts[parts.length-1]+']['+counter);
-            jQuery(proto)
-                .insertBefore(jQuery(this).parent())
-                .trigger('sonata-admin-append-form-element')
-            ;
-
-            jQuery(this).trigger('sonata-collection-item-added');
+            const parts = container.attr("id").split("_");
+            const nameRegexp = new RegExp(`${parts[parts.length - 1]}\\]\\[${protoName}`, "g");
+            proto = proto.replace(nameRegexp, `${parts[parts.length - 1]}][${counter}`);
+            jQuery(proto).
+                insertBefore(jQuery(this).parent()).
+                trigger("sonata-admin-append-form-element");
+            jQuery(this).trigger("sonata-collection-item-added");
+
         });
 
-        jQuery(subject).on('click', '.sonata-collection-delete', function(event) {
+        jQuery(subject).on("click", ".sonata-collection-delete", function (event) {
+
             Admin.stopEvent(event);
 
-            jQuery(this).trigger('sonata-collection-item-deleted');
+            jQuery(this).trigger("sonata-collection-item-deleted");
 
-            jQuery(this).closest('.sonata-collection-row').remove();
+            jQuery(this).closest(".sonata-collection-row").
+                remove();
+
+            jQuery(document).trigger("sonata-collection-item-deleted-successful");
 
-            jQuery(document).trigger('sonata-collection-item-deleted-successful');
         });
+
     },
 
-    setup_per_page_switcher: function(subject) {
-        Admin.log('[core|setup_per_page_switcher] setup page switcher', subject);
+    "setup_per_page_switcher" (subject) {
+
+        Admin.log("[core|setup_per_page_switcher] setup page switcher", subject);
+
+        jQuery("select.per-page").change(function (event) {
 
-        jQuery('select.per-page').change(function(event) {
-            jQuery('input[type=submit]').hide();
+            jQuery("input[type=submit]").hide();
+
+            window.top.location.href = this.options[this.selectedIndex].value;
 
-            window.top.location.href=this.options[this.selectedIndex].value;
         });
+
     },
 
-    setup_form_tabs_for_errors: function(subject) {
-        Admin.log('[core|setup_form_tabs_for_errors] setup form tab\'s errors', subject);
+    "setup_form_tabs_for_errors" (subject) {
```

Comment:
- eslint: Expected object keys to be in ascending order. 'setup_form_tabs_for_errors' should be before 'setup_per_page_switcher'.